project-base/src/Shopsys/ShopBundle/Resources/scripts/frontend/product/productList.js

```diff
@@ -16,10 +16,13 @@
     });
 
     $(document).ready(function () {
-        var ajaxMoreLoader = new Shopsys.productList.AjaxMoreLoader();
-        ajaxMoreLoader.init();
-        var ajaxFilter = new Shopsys.productList.AjaxFilter(ajaxMoreLoader);
-        ajaxFilter.init();
+        $('.js-list-with-paginator').each(function () {
```

Comment:
This seems like a general creation of all AjaxMoreLoaders, but it's in productList.js and it also initializes a Shopsys.productList.AjaxFilter, which is product-specific. This could lead to errors.

I see two possible routes here:
1) Move the general initialization of AjaxMoreLoaders to ajaxMoreLoader.js, leave only AjaxFilter here and get the instance of AjaxMoreLoader for it somehow.
2) Rename the class back to js-product-list-with-paginator and let this javascript handle just the product list and modify the cookbook so you have to create your own (pretty simple) initialization script. This is imho preferable and it will allow you to pass the translated message (which would be a problem to pass though a data-attribute because of how JsTranslatorCompilerPass works) using a anonymous function (something like `function (number) { return Shopsys.translator.transChoice('{1}Load next %loadNextCount% product|[2,Inf]Load next %loadNextCount% products', number) }`)